scenario.py

```diff
@@ -77,24 +78,47 @@ def __init__(self, params, experiment_path):
         self.contributivity_list = []
 
         # Number of epochs and mini-batches in ML training
-        if 'epoch_count' in params.keys():
+        if 'epoch_count' in params:
             self.epoch_count = params['epoch_count']
             assert(self.epoch_count > 0)
         else:
             self.epoch_count = 40
         
-        if 'minibatch_count' in params.keys():
-            self.minibatch_count = params['minibatch_count']
-            assert(self.minibatch_count > 0)
-        else:
-            self.minibatch_count = 20
-
-        if 'minibatch_count' in params.keys():
+        if 'minibatch_count' in params:
             self.minibatch_count = params['minibatch_count']
             assert self.minibatch_count > 0
         else:
             self.minibatch_count = 20
-
+            
+        # Contributivity methods
+        known_methods_list = [
+            "Shapley values",
+            "Independant scores", 
+            "TMCS",
+            ]
+        
+        # List of Contributivity methods runned by default if no method was given in the config file 
+        methods_default = [
+            "Shapley values",
+            "Independant scores",
+            "TMCS values"
+            ]
+        
+        self.methods = []
+        if not 'methods' in params:
+            self.methods = methods_default
+        else:
+            if params['methods']:
+                for el in params['methods']:
+                    if el in known_methods_list:
+                        self.methods.append(el)
+                    else:
+                        raise Exception('method ' + el + ' is not in methods list.')
+            else:
+                raise Exception("No contributivity method given in config file")      
```

Comment:
What do you think of something like this:
```
ALL_METHODS_LIST =  = [...]
DEFAULT_METHODS_LIST = [...]

if 'methods' in params:

	for method in params['methods']:
			if method in ALL_METHODS_LIST:
				self.methods.append(method)
			else:
				raise Exception('method ' + method + ' is not in methods list: ' + ALL_METHODS_LIST)
				
else:
	self.methods = DEFAULT_METHODS_LIST
```

I choose to put 2 methods list (DEFAULT and ALL) because I guess we might not want to run all methods by default.

Reply:
To me that's what is currently implemented (except the `if ... else ...` is reversed but it doesn't change anything): `DEFAULT_METHODS_LIST` is `methods_default` and equivalent to your `ALL_METHODS_LIST` is `known_methods_list`. Or what do I miss?

Reply:
I changed the code so it better fits Romain's suggestion

Reply:
@bowni Yes it is a same but in a more elegant way ;) 
@RomainGoussault I update the code! What do you want in `DEFAULT_METHODS_LIST` ? For the moment, do I write the 3 methods we currently have?

Reply:
@RomainGoussault But in fact, in your code, we do not check if a list is given in config file. If we have the field method but nothing is given, it stops running.